core/src/main/scala/caliban/CalibanError.scala

```diff
@@ -102,3 +106,53 @@ private object ErrorCirce {
   }
 
 }
+
+private object ErrorPlayJson {
+  import play.api.libs.json._
+  import play.api.libs.json.Json.toJson
+
+  private def locationToJson(locInfo: Option[LocationInfo]): Map[String, JsValue] =
+    Some(locInfo).collect {
+      case Some(li) => Json.arr(Json.obj("line" -> toJson(li.line), "column" -> toJson(li.column)))
+    }.fold(Map.empty[String, JsValue])(v => Map("locations" -> v))
```

Comment:
I would just make this a `Writes`

```scala
private implicit val locationInfoWrites: Writes[LocationInfo] =
  Json.writes[LocationInfo].transform(Json.arr(_))

```

Reply:
Oh, I just didn't realize that it was written manually for circe to not depend in circe-generic/circe-derivation, and for play macros already included so I can use them.

Thanks for good suggestions, will clean it up :+1: